src/opm/parser/eclipse/EclipseState/TracerConfig.cpp

```diff
@@ -142,4 +160,36 @@ bool TracerConfig::operator==(const TracerConfig& other) const {
     return this->tracers == other.tracers;
 }
 
+std::string TracerConfig::get_unit_string(const UnitSystem& unit_system, const std::string & tracer_kw) const {
+    if (tracer_kw.length() > 4 ) {
+        std::string tracer_name = tracer_kw.substr(4);
+        for (const auto& tracer : tracers) {
+            if (tracer.name == tracer_name) {
+                std::string unit_string(tracer.unit_string);
+                if (tracer.unit_string != "") {
+                    if (tracer_kw[3] == 'R') {
+                       std::string rateName = unit_system.name(Opm::UnitSystem::measure::rate);
+                       std::size_t found = rateName.find('/');
+                       unit_string += rateName.substr(found);
+                    }
+                    else if (tracer_kw[3] == 'T') {
+                    }
+                    else if (tracer_kw[3] == 'C') {
+                        unit_string += "/";
+                        if (tracer.phase == Phase::GAS )
+                            unit_string += unit_system.name(Opm::UnitSystem::measure::gas_surface_volume);
+                        else /* OIL or WAT */
+                            unit_string += unit_system.name(Opm::UnitSystem::measure::liquid_surface_volume);
+                    }
+                    else {
+                        throw std::runtime_error("Tracer summary kw not recognized: " + tracer_kw);
+                    }
+                }
+                return unit_string;
+            }
+        }
+    }
```

Comment:
I suppose we might switch our approach here to reduce the nesting a bit.  On the other hand, the current code clearly does what it needs to do so I'll hold off on that until a later refactoring.
```C++
using M = UnitSystem::measure;

if (tracer_kw.length() <= 4) {
    return {};
}
const auto tracer_name = tracer_kw.substr(4);
auto tracerPos = std::find_if(this->tracers.begin(), this->tracers.end(),
    [&tracer_name](const auto& tracer)
{
    return tracer.name == tracer_name;
});
if ((tracerPos == this->tracers.end()) || tracerPos->unit_string.empty()) {
    return {};
}

auto unit_string = tracerPos->unit_string;
if (tracer_kw[3] == 'R') {
    const auto rateName = std::string_view(unit_system.name(M::rate));
    unit_string += rateName.substr(rateName.find('/'));
}
else if (tracer_kw[3] == 'C') {
    const auto volume_unit = (tracerPos->phase == Phase::GAS)
        ? M::gas_surface_volume
        : M::liquid_surface_volume;
    unit_string.append('/').append(unit_system.name(volume_unit));
}
else if (tracer_kw[3] != 'T') {
    throw std::runtime_error {
        "Tracer summary keyword '" + tracer_kw + "' is not recognized"
    };
}

return unit_string;
```